control-plane/connect-inject/common/common.go

```diff
@@ -177,3 +180,19 @@ func PortValueFromIntOrString(pod corev1.Pod, port intstr.IntOrString) (uint32,
 	}
 	return uint32(portVal), nil
 }
+
+// ConsulNamespaceIsNotFound checks the gRPC error code and message to determine if the
+// error indicates that the namespace of for a resource does not yet exist.
+func ConsulNamespaceIsNotFound(err error) bool {
+	if err == nil {
+		return false
+	}
+	s, ok := status.FromError(err)
+	if !ok {
+		return false
+	}
+	if codes.InvalidArgument == s.Code() && strings.Contains(s.Message(), "namespace resource not found") {
```

Comment:
Do we need to verify the message? Or would checking for the `InvalidArgument` code be sufficient?

Reply:
> Do we need to verify the message? Or would checking for the InvalidArgument code be sufficient?

I wouldn't rely just on that code in particular to identify "missing namespace" since it can show up in a number of cases - though TBF, delete is more limited than most. We could end up misreporting the actual reason for failure instead of propagating the underlying error.

Though, as a side note, it's probably worth including the original error message in the log @DanStough so we can at least catch a misidentification bug if it occurs.

Reply:
❤️ 
* Add original error message
* Add integration test with the server to test the message.

Reply:
@zalimeni added the Int test. Let me know if that's what you were thinking. We would also catch it on acceptance tests if Consul changed.